pkgs/top-level/stage.nix

```diff
@@ -8,7 +8,15 @@
    arguments. Normal users should not import this directly but instead
    import `pkgs/default.nix` or `default.nix`. */
 
-
+let
+  # Cache mapping from attributes to unit directories up here so it can get
+  # reused every time this file is called
+  unitDirs = builtins.foldl' (result: shard:
+    result // builtins.mapAttrs (name: type:
+      ../unit + "/${shard}/${name}/pkg-fun.nix"
```

Comment:
This implies a hardcoded limit of 2 in the subdirs?

Reply:
Yes indeed, two levels means that `pkgs/unit` has a bit less than 1000 subdirectories, and each of those also has less than 1000 subdirectories with the current package distribution.